pageserver/src/tenant/timeline.rs

```diff
@@ -161,6 +161,10 @@ impl<T: AsLayerDesc + ?Sized> LayerFileManager<T> {
         }
     }
 
+    pub(crate) fn contains(&self, layer: &Arc<T>) -> bool {
+        self.0.contains_key(&layer.layer_desc().key())
+    }
```

Comment:
Like I was trying to sell earlier with Konstantin's PRs on the same topic, I would prefer the `insert` method returning a bool, like `HashSet::insert` (was a mapping modified) instead of asking before doing the action. I am unsure if it makes sense to have the option of allowing this contains check (if we remember it) to affect the decision. I have been thinking it'd be best if this was an internal thing... But now I remember that the *file itself* has already been renamed over by DeltaFileWriter.

Well, let's just offer this as a note while I recap the other alternatives.

Reply:
I would prefer doing it as is so that `insert` will panic on inconsistency. I think we are using `insert_historic` everywhere and don't want to check the return value at every call site.

Reply:
I think this will lead to an inconsistent handling, but let's see if it will. Resolving.